src/Build/BackEnd/Components/Logging/TargetLoggingContext.cs

```diff
@@ -105,7 +90,7 @@ internal void LogTargetBatchFinished(string projectFullPath, bool success, IEnum
             TargetOutputItemsInstanceEnumeratorProxy targetOutputWrapper = null;
 
             // Only log target outputs if we are going to log a target finished event and the environment variable is set and the target outputs are not null
-            if (!LoggingService.OnlyLogCriticalEvents && s_enableTargetOutputLogging && targetOutputs != null)
+            if (!LoggingService.OnlyLogCriticalEvents && LoggingService.EnableTargetOutputLogging && targetOutputs != null)
```

Comment:
This is the core change necessary to make TL super-powered. We need a way for the entire LoggingService to contain the opt-in so that logger configuration can control behavior instead of magic env vars.